parser/generic_parser.go

```diff
@@ -101,7 +99,24 @@ func (rp *GenericParser) ParseHeader(tr *tar.Reader, hdr *tar.Header, hPath stri
 	return nil
 }
 
+// Default data parser what writes the uncompressed data to `w`.
 func parseData(r io.Reader, w io.Writer, uFiles map[string]UpdateFile) error {
+	return parseDataWithHandler(r,
+		func(dr io.Reader, uf UpdateFile) error {
+			if _, err := io.Copy(w, dr); err != nil {
+				return errors.Wrapf(err, "parser: can not read data: %v", uf.Name)
+			}
+			return nil
+		},
+		uFiles)
+}
+
+// Caller provided update data stream handler. Parameter `r` is a decompressed
+// data stream, `uf` contains basic information about update. The handler shall
+// return nil if no errors occur.
+type parseDataHandlerFunc func(r io.Reader, uf UpdateFile) error
```

Comment:
I think this should be exported. This will make it more obvious that you need to implement this yourself to use wit some custom parser type.